test/e2e/serial/tests/workload_overhead.go

```diff
@@ -274,6 +274,158 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload overhea
+
+				// even if it is not a hard rule, and even if there are a LOT of edge cases, a good starting point is usually
+				// in the ballpark of 5x the base load. We start like this
+				podResources := corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("6"),
+					corev1.ResourceMemory: resource.MustParse("6Gi"),
+				}
+
+				// to avoid issues with fractional resources being unaccounted atm, we round up to requests;
+				// for the test proper, as low as cpu=100m and mem=100Mi would have been sufficient.
+				minRes := corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("1"),
+					corev1.ResourceMemory: resource.MustParse("1Gi"),
+				}
+
+				// need a zone with resources for overhead, pod and a little bit more to avoid zone saturation
+				klog.Infof("kubernetes pod fixed overhead: %s", e2ereslist.ToString(rtClass.Overhead.PodFixed))
+
+				const requiredNUMAZones = 2
+				By(fmt.Sprintf("filtering available nodes with at least %d NUMA zones", requiredNUMAZones))
+				nrtCandidates := e2enrt.FilterZoneCountEqual(nrts, requiredNUMAZones)
+
+				const neededNodes = 1
+				if len(nrtCandidates) < neededNodes {
+					Skip(fmt.Sprintf("not enough nodes with at least %d NUMA Zones: found %d, needed %d", requiredNUMAZones, len(nrtCandidates), neededNodes))
+				}
+				candidateNodeNames := e2enrt.AccumulateNames(nrtCandidates)
+
+				var ok bool
+				targetNodeName, ok = e2efixture.PopNodeName(candidateNodeNames)
+				Expect(ok).To(BeTrue(), "cannot select a node among %#v", candidateNodeNames.List())
+				By(fmt.Sprintf("selecting node to schedule the test pod: %q", targetNodeName))
+
+				err = fxt.Client.List(context.TODO(), &targetNrtListInitial)
+				Expect(err).ToNot(HaveOccurred())
+				targetNrtInitial, err = e2enrt.FindFromList(targetNrtListInitial.Items, targetNodeName)
+				Expect(err).NotTo(HaveOccurred())
+
+				//get maximum available node CPU and Memory
+				rtClass.Overhead = &nodev1.Overhead{
+					PodFixed: corev1.ResourceList{
+						corev1.ResourceCPU:    allocatableResourceType(*targetNrtInitial, corev1.ResourceCPU),
+						corev1.ResourceMemory: allocatableResourceType(*targetNrtInitial, corev1.ResourceMemory),
+					},
+				}
+
+				// updating the runtimeClass with the new podoverhead value
+				err = fxt.Client.Update(context.TODO(), rtClass)
+				Expect(err).NotTo(HaveOccurred())
+
+				podFixedOverheadCPU, podFixedOverheadMem := resourcelist.RoundUpCoreResources(*rtClass.Overhead.PodFixed.Cpu(), *rtClass.Overhead.PodFixed.Memory())
+				podFixedOverhead := corev1.ResourceList{
+					corev1.ResourceCPU:    podFixedOverheadCPU,
+					corev1.ResourceMemory: podFixedOverheadMem,
+				}
+				klog.Infof("kubernetes pod fixed overhead rounded to: %s", e2ereslist.ToString(podFixedOverhead))
+
+				zoneRequiredResources := podResources.DeepCopy()
+				resourcelist.AddCoreResources(zoneRequiredResources, podFixedOverhead)
+				resourcelist.AddCoreResources(zoneRequiredResources, minRes)
```

Comment:
we want to have the total pod + overhead resources allocatable under at least one numa zone of the target node but not available.
having the overhead as the max res of the target node and then rounding it up + the pod resources will end up with the pod pending but no guarantee why, it might be because pod + overhead res are not allocatable (there are no place on the node for the pod + OH because the node might be rejected by Fit as Talor mentioned)

Reply:
Sure, I might have misunderstood the description of this test in test plan. I will update this now.